lib/handlers/router-entities/route-caching/dynamo-route-caching-provider.ts

```diff
@@ -0,0 +1,302 @@
+          // Finally we unmarshal that JSON into a `CachedRoutes` object
+          const cachedRoutes: CachedRoutes = CachedRoutesMarshaller.unmarshal(cachedRoutesJson)
+
+          log.info({ cachedRoutes }, `[DynamoRouteCachingProvider] Returning the cached and unmarshalled route.`)
+
+          return cachedRoutes
+        } else {
+          log.info(`[DynamoRouteCachingProvider] No items found in the query response.`)
+        }
+      } catch (error) {
+        log.error({ queryParams, error }, `[DynamoRouteCachingProvider] Error while fetching route from cache`)
+      }
+    }
+
+    // We only get here if we didn't find a cachedRoutes
+    return undefined
+  }
+
+  /**
+   * Implementation of the abstract method defined in `IRouteCachingProvider`
+   * Attempts to insert the `CachedRoutes` object into cache, if the CachingStrategy returns the CachingParameters
+   *
+   * @param cachedRoutes
+   * @param amount
+   * @protected
+   */
+  protected async _setCachedRoute(cachedRoutes: CachedRoutes, amount: CurrencyAmount<Currency>): Promise<boolean> {
+    const cachedRoutesStrategy = this.getCachedRoutesStrategyFromCachedRoutes(cachedRoutes)
+    const cachingParameters = cachedRoutesStrategy?.getCachingParameters(amount)
+
+    if (cachingParameters) {
+      // TTL is ttl_minutes from now. multiply ttl_minutes times 60 to convert to seconds, since ttl is in seconds.
+      const ttl = Math.floor(Date.now() / 1000) + 60 * this.ttl_minutes
+      // Marshal the CachedRoutes object in preparation for storing in DynamoDB
+      const marshalledCachedRoutes = CachedRoutesMarshaller.marshal(cachedRoutes)
+      // Convert the marshalledCachedRoutes to JSON string
+      const jsonCachedRoutes = JSON.stringify(marshalledCachedRoutes)
+      // Encode the jsonCachedRoutes into Binary
+      const binaryCachedRoutes = Buffer.from(jsonCachedRoutes)
+
+      const putParams = {
+        TableName: this.tableName,
+        Item: {
+          pairTradeTypeChainId: `${cachedRoutes.tokenIn.address}/${cachedRoutes.tokenOut.address}/${cachedRoutes.tradeType}/${cachedRoutes.chainId}`,
+          protocolsBucketBlockNumber: `${cachedRoutes.protocolsCovered}/${cachingParameters.bucket}/${cachedRoutes.blockNumber}`,
+          item: binaryCachedRoutes,
+          ttl: ttl,
+        },
+      }
+
+      log.info(
+        { putParams, cachedRoutes, jsonCachedRoutes },
+        `[DynamoRouteCachingProvider] Attempting to insert route to cache`
+      )
+
+      try {
+        await this.ddbClient.put(putParams).promise()
+        log.info(`[DynamoRouteCachingProvider] Cached route inserted to cache`)
+
+        return true
+      } catch (error) {
+        log.error({ error, putParams }, `[DynamoRouteCachingProvider] Cached route failed to insert`)
+
+        return false
+      }
+    } else {
+      // No CachingParameters found, return false to indicate the route was not cached.
+
```

Comment:
I don't think we need to log something here, this will be the "main" case since the CachingParameters are in the configuration file, I think that these logs will be unnecessary